### PR TITLE
fix(building): make default building entrance editable

### PR DESCRIPTION
**Pull request by luytena**

### CODE REVIEW

File: addon/components/link-street.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import Component from "@glimmer/component";
+import Models from "ember-ebau-gwr/models";
+
+export default class LinkStreetComponent extends Component {
+  Models = Models;
```
Comment (velrest) [inline addon/components/link-street.js:5]:
```suggestion
import Street from "ember-ebau-gwr/models/street";

export default class LinkStreetComponent extends Component {
  Street = Street;
```

File: addon/controllers/building/edit/entrance/edit/link-street.js [hunk 2]
```diff
@@ -18,11 +19,22 @@ export default class BuildingEditEntranceLinkStreetController extends Controller
...
+
     return `building.edit.entrance.${
       this.buildingEntrance.newRecord ? "new" : "edit.index"
     }`;
   }
 
+  get backRouteLabel() {
+    return this.intl.t(this.model.buildingId === "new"
+      ? "ember-gwr.buildingEntrance.backToBuilding"
+      : "ember-gwr.buildingEntrance.backToEntrance");
```
Comment (velrest) [inline addon/controllers/building/edit/entrance/edit/link-street.js:35]:
```suggestion
    return this.intl.t(`ember-gwr.buildingEntrance.${
      this.model.buildingId === "new"
        ? "backToBuilding"
        : "backToEntrance"
      }`);
```

File: addon/controllers/building/edit/form.js [hunk 3]
```diff
@@ -38,7 +39,10 @@ export default class BuildingFormController extends Controller {
       yield this.fetchBuilding.perform();
 
       if (this.model.buildingWork?.isNew) {
-        return this.model.buildingWork;
+        console.log("this.building.newRecord:", this.buildingAPI.newRecord);
+        return this.buildingAPI.newRecord
+          ? this.buildingAPI.newRecord
+          : this.model.buildingWork;
```
Comment (velrest) [inline addon/controllers/building/edit/form.js:45]:
```suggestion
        return this.buildingAPI.newRecord ?? this.model.buildingWork;
```

File: addon/controllers/building/edit/form.js [hunk 4]
```diff
@@ -217,4 +221,10 @@ export default class BuildingFormController extends Controller {
   getChangeHint(currentStatus, newStatus) {
     return this.buildingAPI.getChangeHint(currentStatus, newStatus);
   }
+
+  @action
+  updateEntrance(key, value) {
+    console.log("kv:", key, value);
+    set(this.buildingWork.building.buildingEntrance[0], key, value);
```
Comment (velrest) [inline addon/controllers/building/edit/form.js:228]: Maybe comment about key beeing a dot-propertybalbal

File: addon/models/building-entrance.js [hunk 5]
```diff
@@ -55,12 +55,14 @@ export default class BuildingEntrance extends XMLModel {
...
   static template = `
   {{#if model.isNew}}
-    {{#if model.buildingEntranceNo}}
-      <ns2:buildingEntranceNo>{{model.buildingEntranceNo}}</ns2:buildingEntranceNo>
-    {{/if}}
-    <ns2:isOfficialAddress>{{model.isOfficialAddress}}</ns2:isOfficialAddress>
-    {{> Street model=model.street}}
-    {{> Locality model=model.locality}}
+    <ns2:buildingEntrance>
+      {{#if model.buildingEntranceNo}}
```
Comment (velrest) [inline addon/models/building-entrance.js:59]: `{{{modelField...}}}`

File: addon/models/building-entrance.js [hunk 6]
```diff
@@ -55,12 +55,14 @@ export default class BuildingEntrance extends XMLModel {
 
   static template = `
   {{#if model.isNew}}
-    {{#if model.buildingEntranceNo}}
-      <ns2:buildingEntranceNo>{{model.buildingEntranceNo}}</ns2:buildingEntranceNo>
-    {{/if}}
-    <ns2:isOfficialAddress>{{model.isOfficialAddress}}</ns2:isOfficialAddress>
-    {{> Street model=model.street}}
-    {{> Locality model=model.locality}}
+    <ns2:buildingEntrance>
```
Comment (velrest) [inline addon/models/building-entrance.js:58]: Api doesnt like `<ns2:buildingEntrance` on most request types

File: addon/models/building.js [hunk 7]
```diff
@@ -235,14 +237,27 @@ export default class Building extends XMLModel {
...
-      <ns2:buildingEntrance>
-        <ns2:buildingEntranceNo>4</ns2:buildingEntranceNo>
-        <ns2:locality>
-          <ns2:name>
-            <ns2:nameLong>{{model.municipalityName}}</ns2:nameLong>
-          </ns2:name>
-        </ns2:locality>
-      </ns2:buildingEntrance>
+      {{#each model.buildingEntrance}}
+        {{> BuildingEntrance model=this}}
```
Comment (velrest) [inline addon/models/building.js:241]: Please comment :smile:

File: addon/routes/building/edit/index.js [hunk 8]
```diff
@@ -1,3 +1,13 @@
...
+import { inject as service } from "@ember/service";
 
-export default class BuildingEditIndexRoute extends Route {}
+export default class BuildingEditIndexRoute extends Route {
+  @service building;
+
+  model() {
+    // reset the newRecord
+    console.log("model buidling");
+    this.building.newRecord = null;
```
Comment (velrest) [inline addon/routes/building/edit/index.js:10]: Would be nice if this were in `beforeModel` since the model hooks is just for "fetching" model data

**Review by czosel — APPROVED** [review]

**Pull request merged**